src/script/script.cpp

```diff
@@ -328,3 +331,11 @@ bool GetScriptOp(CScriptBase::const_iterator& pc, CScriptBase::const_iterator en
     opcodeRet = static_cast<opcodetype>(opcode);
     return true;
 }
+
+bool IsOpSuccess(const opcodetype& opcode)
+{
+    return opcode == 80 || opcode == 98 || (opcode >= 126 && opcode <= 129) ||
+           (opcode >= 131 && opcode <= 134) || (opcode >= 137 && opcode <= 138) ||
+           (opcode >= 141 && opcode <= 142) || (opcode >= 149 && opcode <= 153) ||
+           (opcode >= 187 && opcode <= 254);
```

Comment:
332dc13b maybe simplify
```diff
     return opcode == 80 || opcode == 98 || (opcode >= 126 && opcode <= 129) ||
-           (opcode >= 131 && opcode <= 134) || (opcode >= 137 && opcode <= 138) ||
-           (opcode >= 141 && opcode <= 142) || (opcode >= 149 && opcode <= 153) ||
+           (opcode >= 131 && opcode <= 134) || opcode == 137 || opcode == 138 ||
+           opcode == 141 || opcode == 142 || (opcode >= 149 && opcode <= 153) ||
            (opcode >= 187 && opcode <= 254);
```

Reply:
IIRC the text as-is completely "matches" the BIP text, meaning if it gives a range here, it gives a range in the BIP text. I think as-is is easier to pattern match.

Reply:
True, though the BIP prose here is inconsistent (see the first 2 elements) and could be updated. The code could opt for the simpler version. Not a biggie tho :)

Reply:
I prefer to keep it exactly as in the BIP for now, but no reason the BIP can't be updated to be more clear.